packages/core/src/components/ImageViewer/ZoomControls.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useCallback } from 'react';
+import IconAdd from '@airbnb/lunar-icons/src/interface/IconAdd';
+import IconRemove from '@airbnb/lunar-icons/src/interface/IconRemove';
+import Button from '../Button';
+import IconButton from '../IconButton';
+import ButtonGroup from '../ButtonGroup';
+import Dropdown from '../Dropdown';
+import Menu, { Item } from '../Menu';
+import T from '../Translate';
+import useStyles, { StyleSheet } from '../../hooks/useStyles';
+
+const styleSheet: StyleSheet = () => ({
+  controls: {
+    position: 'relative',
+    display: 'inline-block',
+  },
+});
+
+const ZOOM_FACTOR = 0.5;
+const ZOOM_OPTIONS = [
+  {
+    label: '100%',
+    scale: 1,
+  },
+  {
+    label: '125%',
+    scale: 1.25,
+  },
+  {
+    label: '150%',
+    scale: 1.5,
+  },
+  {
+    label: '200%',
+    scale: 2,
+  },
+];
+
+export type Props = {
+  /** The current scale / zoom level. 1 by default. */
+  scale?: number;
+  /** Callback when scale / zoom changes */
+  onScale: (scale: number) => void;
+};
+
+/** Zoom controls that can be used with an image viewer component */
+export default function ZoomControls(props: Props) {
+  const [styles, cx] = useStyles(styleSheet);
+  const [visible, setVisible] = useState(false);
+  const { onScale, scale = 1 } = props;
+
+  function getItemOnClick(value: number) {
+    return useCallback(() => {
+      onScale(value);
+      setVisible(false);
+    }, []);
+  }
+
+  const zoomOptions = ZOOM_OPTIONS.map((zoom: { label: string; scale: number }) => ({
+    ...zoom,
+    handleOnClick: getItemOnClick(zoom.scale),
+  }));
+
+  const handleZoomOut = useCallback(
+    () => onScale(scale - ZOOM_FACTOR < 1 ? 1 : scale - ZOOM_FACTOR),
+    [scale],
+  );
+  const handleZoomIn = useCallback(() => onScale(scale + ZOOM_FACTOR), [scale]);
+  const toggleZoomMenu = useCallback(() => setVisible(!visible), [visible]);
+
+  return (
+    <div className={cx(styles.controls)}>
+      <ButtonGroup>
+        <IconButton onClick={handleZoomOut} disabled={scale === 1}>
+          <IconRemove
+            accessibilityLabel={T.phrase('Zoom out', {}, 'Label for zoom out button')}
+            size="2em"
+          />
+        </IconButton>
+        <Button borderless onClick={toggleZoomMenu}>
```

Comment:
one thing that really helps me with readability is adding new lines between sibling components/tags. breaks up the dense return a bit:

```jsx
      <ButtonGroup>
        <IconButton onClick={handleZoomOut} disabled={scale === 1}>
          <IconRemove
            accessibilityLabel={T.phrase('Zoom out', {}, 'Label for zoom out button')}
            size="2em"
          />
        </IconButton>

        <Button borderless onClick={toggleZoomMenu}>
          {scale * 100}%
        </Button>

        <IconButton onClick={handleZoomIn}>
          <IconAdd
            accessibilityLabel={T.phrase('Zoom in', {}, 'Label for zoom in button')}
            size="2em"
          />
        </IconButton>
      </ButtonGroup>
      
      {visible && (
        <Dropdown visible={visible} left="0" onClickOutside={toggleZoomMenu} zIndex={5}>
          <Menu
            accessibilityLabel={T.phrase(
              'Zoom dropdown menu',
              {},
              'Label for dropdown menu with zoom options',
            )}
          >
            {zoomOptions.map(zoom => (
              <Item key={zoom.scale} onClick={zoom.handleOnClick}>
                {zoom.label}
              </Item>
            ))}
          </Menu>
        </Dropdown>
      )}
```